### ISSUE TITLE
Incorrect BC build failure for METHOD_RETURN_TYPE_CHANGED private to protected

### ISSUE THREAD

**Issue by garydgregory** [siom79/japicmp#265]
JApiCmp raises a false METHOD_RETURN_TYPE_CHANGED to break BC when a `private` method is made `protected`.

Over at Apache Commons Text, this commit: https://github.com/apache/commons-text/commit/12428582b46e9e3ce0dff65e25c51a71083c19b7 broke our build on BC check.

The commit changed the `org.apache.commons.text.StringSubstitutor` version 1.8 method signature from:
```
private int substitute(final TextStringBuilder builder, final int offset, final int length,
            List<String> priorVariables)
```
to (1.9-SNAPSHOT):
```
protected Result substitute(final TextStringBuilder builder, final int offset, final int length,
        List<String> priorVariables)
```
Which does not break BC but JApiCmp fails the build nonetheless:
```
[ERROR] Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.14.3:cmp (default-cli) on project commons-text: There is at least one incompatibility: org.apache.commons.text.StringSubstitutor.substitute(org.apache.commons.text.TextStringBuilder,int,int,java.util.List):METHOD_RETURN_TYPE_CHANGED -> [Help 1]
```
We use:
```
<commons.japicmp.breakBuildOnBinaryIncompatibleModifications>true</commons.japicmp.breakBuildOnBinaryIncompatibleModifications>
<commons.japicmp.version>0.14.3</commons.japicmp.version>
```
JApiCmp should not fail a BC check when a private method becomes protected (or public) when its signature changes.

**Comment by scordio** [comment]
Relates to #384.

**Comment by siom79** [comment]
Fixed with 0.19.0.
